src/datatable.js

```diff
@@ -0,0 +1,241 @@
+  columnsArray    = [];
+  sortingCriteria = {};
+  searchCriteria  = {}
+
+  constructor(Router, element) {
+    this.router  = Router;
+    this.element = element;
+  }
+
+  attached() {
+    this.load();
+  }
+
+  load() {
+    this.criteria = this.buildCriteria();
+
+    this.repository.find(this.criteria, true).then(result => {
+     this.data = result;
+    })
+    .catch(error => {
+      console.error('Something went wrong.', error);
+    });
+  }
+
+  pageChanged(newValue, oldvalue) {
+    this.load();
+  }
+
+  buildCriteria() {
+    let crit = {};
+
+    if (this.searchable !== null && Object.keys(this.searchCriteria).length ) {
+      let propertyName = Object.keys(this.searchCriteria)[0];
+      if (this.searchCriteria[propertyName]) {
+        crit['where']                           = {};
+        crit['where'][propertyName]             = {};
+        crit['where'][propertyName]['contains'] = this.searchCriteria[propertyName];
+
+        this.criteriaPager = crit['where'];
+      }
+    }
+
+    if (this.sortable !== null && Object.keys(this.sortingCriteria).length ) {
+      let propertyName = Object.keys(this.sortingCriteria)[0];
+      if (this.sortingCriteria[propertyName]) {
+        crit['sort'] = propertyName + ' ' + this.sortingCriteria[propertyName];
+      }
+    }
+
+    crit['skip']  = this.page * this.limit - this.limit;
+    crit['limit'] = this.limit;
+
+    return crit;
+  }
+
+  populate (row) {
+    return this.repository.getPopulatedEntity(row);
+  }
+
+  doDelete(row) {
+    if (typeof this.delete === 'function') {
+      return this.delete(this.populate(row));
+    }
+
+    this.populate(row).destroy()
+      .then(ah => {
+      this.load();
+      this.triggerEvent('deleted', row);
+    })
+    .catch(error => {
+      this.triggerEvent('exception', {on: 'delete', error: error});
+    });
+  }
+
+  doUpdate (row) {
+    if (typeof this.update === 'function') {
+      return this.update(this.populate(row));
+    }
+
+    this.populate(row).update()
+      .then(() => {
+      this.load();
+      this.triggerEvent('updated', row);
+    })
+    .catch(error => {
+      this.triggerEvent('exception', {on: 'update', error: error});
+    });
+  }
+
+  doSort(columnLabel) {
+    if (this.sortable === null || this.isObject(columnLabel.column)) {
+      return;
+    }
+
+    if (this.sortingCriteria[columnLabel.column]) {
+      this.sortingCriteria[columnLabel.column] = (this.sortingCriteria[columnLabel.column] === 'asc' ? 'desc' : 'asc');
+    }
+    else {
+      this.sortingCriteria = {};
+      this.sortingCriteria[columnLabel.column] = 'asc';
+    }
+
+    this.load();
+  }
+
+  doSearch(searchInput) {
+    if (this.searchable === null) {
+      return;
+    }
+
+    if (!(this.defaultColumn in this.searchCriteria)) {
+      this.searchCriteria = {};
+    }
+
+    this.searchCriteria[this.defaultColumn] = searchInput;
+
+    this.load();
+  }
+
+  @computedFrom('columns')
+  get columnLabels () {
```

Comment:
Usually I'd agree, but for now I think it's best to leave it like this. It works, it's clear. We'll make this pretty later :)